pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -585,3 +636,9 @@ func (k *Kaniko) deleteJob(namespace string, jobName string) error {
 	k.logger.DebugWith("Successfully deleted job", "namespace", namespace, "job", jobName)
 	return nil
 }
+
+func (k *Kaniko) matchECRRegex(registryURL string) bool {
+	ecrRegex := regexp.MustCompile(
+		`([a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com`)
+	return ecrRegex.MatchString(registryURL)
```

Comment:
how about `HasSuffix(registryURL, ".amazonaws.com") && strings.Contains(registryURL, ".ecr.")`? more readable and maintainable